Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -1035,15 +1030,16 @@ class KalmanFitter {
     // To be able to find measurements later, we put them into a map
     // We need to copy input SourceLinks anyways, so the map can own them.
     ACTS_VERBOSE("Preparing " << sourcelinks.size() << " input measurements");
-    std::map<const Surface*, source_link_t> inputMeasurements;
+    std::map<GeometryIdentifier, source_link_t> inputMeasurements;
```

Comment:
Oh, is that even correct? Later on you still pass in surface pointers. Are they auto converted? If I'm reading this right, this is dangerous. Also, not the point of this PR, right?

Reply:
That should of course be `const Surface*`.  This was a left-over from a change that is not part of this PR anymore. I am surprised that this was compiling. Fixed now.